src/mono/mono/metadata/sre.c

```diff
@@ -835,32 +781,11 @@ method_encode_methodspec (MonoDynamicImage *assembly, MonoMethod *method)
 	imethod = (MonoMethodInflated *) method;
 	declaring = imethod->declaring;
 
-	sig = mono_dynimage_encode_method_signature (assembly, mono_method_signature_internal (declaring));
-	mtoken = mono_image_get_memberref_token (assembly, m_class_get_byval_arg (method->klass), declaring->name, sig);
+	mtoken = mono_image_get_memberref_token (assembly, m_class_get_byval_arg (method->klass), declaring->name, 0);
```

Comment:
`mono_image_get_memberref_token` is `static` in this file.  You can just change the signature of it and of `mono_image_add_memberef_row` (also static) not to take a sig token argument.